internals/overlord/logstate/gatherer.go

```diff
@@ -312,6 +314,10 @@ type logClient interface {
 
 	// Flush sends buffered logs (if any) to the remote target.
 	Flush(context.Context) error
+
+	// AddEnv adds a service's environment to the client, so it can correctly
+	// evaluate the labels defined in the plan.
+	AddEnv(serviceName string, env []string)
```

Comment:
don't know about "need" but I guess it's a reasonable addition

Reply:
@hpidcock maybe we should make these methods more abstract, e.g. `ServiceAdded` and `ServiceRemoved`? We might have more things in future that need to be passed in on service start/shutdown, so we could just extend these methods instead of adding new ones

Reply:
Just for the record after our spec review meeting, we decided this was the right place for this, but:

* It would be about labels rather than environment
* The method would be `SetLabels(serviceName string, labels map[string]string)` method
* Labels=nil would mean remove this service